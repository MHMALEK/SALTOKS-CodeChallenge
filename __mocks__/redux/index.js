export default function mockRedux() {
  // dispatch event
}
