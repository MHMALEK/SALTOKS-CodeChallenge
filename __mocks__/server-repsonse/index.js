export const repositoriseSearchApi = {
         incomplete_results: false,
         total_count: 24034,
         items: [
           {
             archive_url:
               "https://api.github.com/repos/asdf-vm/asdf/{archive_format}{/ref}",
             archived: false,
             assignees_url:
               "https://api.github.com/repos/asdf-vm/asdf/assignees{/user}",
             blobs_url: "https://api.github.com/repos/asdf-vm/asdf/git/blobs{/sha}",
             branches_url:
               "https://api.github.com/repos/asdf-vm/asdf/branches{/branch}",
             clone_url: "https://github.com/asdf-vm/asdf.git",
             collaborators_url:
               "https://api.github.com/repos/asdf-vm/asdf/collaborators{/collaborator}",
             comments_url:
               "https://api.github.com/repos/asdf-vm/asdf/comments{/number}",
             commits_url: "https://api.github.com/repos/asdf-vm/asdf/commits{/sha}",
             compare_url:
               "https://api.github.com/repos/asdf-vm/asdf/compare/{base}...{head}",
             contents_url:
               "https://api.github.com/repos/asdf-vm/asdf/contents/{+path}",
             contributors_url:
               "https://api.github.com/repos/asdf-vm/asdf/contributors",
             created_at: "2014-11-29T04:41:43Z",
             default_branch: "master",
             deployments_url:
               "https://api.github.com/repos/asdf-vm/asdf/deployments",
             description:
               "Extendable version manager with support for Ruby, Node.js, Elixir, Erlang & more",
             disabled: false,
             downloads_url: "https://api.github.com/repos/asdf-vm/asdf/downloads",
             events_url: "https://api.github.com/repos/asdf-vm/asdf/events",
             fork: false,
             forks: 284,
             forks_count: 284,
             forks_url: "https://api.github.com/repos/asdf-vm/asdf/forks",
             full_name: "asdf-vm/asdf",
             git_commits_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/commits{/sha}",
             git_refs_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/refs{/sha}",
             git_tags_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/tags{/sha}",
             git_url: "git://github.com/asdf-vm/asdf.git",
             has_downloads: true,
             has_issues: true,
             has_pages: true,
             has_projects: true,
             has_wiki: true,
             homepage: "https://asdf-vm.com/",
             hooks_url: "https://api.github.com/repos/asdf-vm/asdf/hooks",
             html_url: "https://github.com/asdf-vm/asdf",
             id: 27291507,
             issue_comment_url:
               "https://api.github.com/repos/asdf-vm/asdf/issues/comments{/number}",
             issue_events_url:
               "https://api.github.com/repos/asdf-vm/asdf/issues/events{/number}",
             issues_url: "https://api.github.com/repos/asdf-vm/asdf/issues{/number}",
             keys_url: "https://api.github.com/repos/asdf-vm/asdf/keys{/key_id}",
             labels_url: "https://api.github.com/repos/asdf-vm/asdf/labels{/name}",
             language: "Shell",
             languages_url: "https://api.github.com/repos/asdf-vm/asdf/languages",
             license: {
               key: "mit",
               name: "MIT License",
               spdx_id: "MIT",
               url: "https://api.github.com/licenses/mit"
             },
             merges_url: "https://api.github.com/repos/asdf-vm/asdf/merges",
             milestones_url:
               "https://api.github.com/repos/asdf-vm/asdf/milestones{/number}",
             mirror_url: null,
             name: "asdf",
             node_id: "MDEwOlJlcG9zaXRvcnkyNzI5MTUwNw==",
             notifications_url:
               "https://api.github.com/repos/asdf-vm/asdf/notifications{?since,all,participating}",
             open_issues: 52,
             open_issues_count: 52,
             owner: {
               login: "asdf-vm",
               id: 17869024,
               node_id: "MDEyOk9yZ2FuaXphdGlvbjE3ODY5MDI0"
             },
             private: false,
             pulls_url: "https://api.github.com/repos/asdf-vm/asdf/pulls{/number}",
             pushed_at: "2019-11-29T16:45:42Z",
             releases_url: "https://api.github.com/repos/asdf-vm/asdf/releases{/id}",
             score: 115.17725,
             size: 862,
             ssh_url: "git@github.com:asdf-vm/asdf.git",
             stargazers_count: 6253,
             stargazers_url: "https://api.github.com/repos/asdf-vm/asdf/stargazers",
             statuses_url:
               "https://api.github.com/repos/asdf-vm/asdf/statuses/{sha}",
             subscribers_url:
               "https://api.github.com/repos/asdf-vm/asdf/subscribers",
             subscription_url:
               "https://api.github.com/repos/asdf-vm/asdf/subscription",
             svn_url: "https://github.com/asdf-vm/asdf",
             tags_url: "https://api.github.com/repos/asdf-vm/asdf/tags",
             teams_url: "https://api.github.com/repos/asdf-vm/asdf/teams",
             trees_url: "https://api.github.com/repos/asdf-vm/asdf/git/trees{/sha}",
             updated_at: "2019-11-30T05:54:54Z",
             url: "https://api.github.com/repos/asdf-vm/asdf",
             watchers: 6253,
             watchers_count: 6253
           },
           {
             archive_url:
               "https://api.github.com/repos/asdf-vm/asdf/{archive_format}{/ref}",
             archived: false,
             assignees_url:
               "https://api.github.com/repos/asdf-vm/asdf/assignees{/user}",
             blobs_url: "https://api.github.com/repos/asdf-vm/asdf/git/blobs{/sha}",
             branches_url:
               "https://api.github.com/repos/asdf-vm/asdf/branches{/branch}",
             clone_url: "https://github.com/asdf-vm/asdf.git",
             collaborators_url:
               "https://api.github.com/repos/asdf-vm/asdf/collaborators{/collaborator}",
             comments_url:
               "https://api.github.com/repos/asdf-vm/asdf/comments{/number}",
             commits_url: "https://api.github.com/repos/asdf-vm/asdf/commits{/sha}",
             compare_url:
               "https://api.github.com/repos/asdf-vm/asdf/compare/{base}...{head}",
             contents_url:
               "https://api.github.com/repos/asdf-vm/asdf/contents/{+path}",
             contributors_url:
               "https://api.github.com/repos/asdf-vm/asdf/contributors",
             created_at: "2014-11-29T04:41:43Z",
             default_branch: "master",
             deployments_url:
               "https://api.github.com/repos/asdf-vm/asdf/deployments",
             description:
               "Extendable version manager with support for Ruby, Node.js, Elixir, Erlang & more",
             disabled: false,
             downloads_url: "https://api.github.com/repos/asdf-vm/asdf/downloads",
             events_url: "https://api.github.com/repos/asdf-vm/asdf/events",
             fork: false,
             forks: 284,
             forks_count: 284,
             forks_url: "https://api.github.com/repos/asdf-vm/asdf/forks",
             full_name: "asdf-vm/asdf",
             git_commits_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/commits{/sha}",
             git_refs_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/refs{/sha}",
             git_tags_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/tags{/sha}",
             git_url: "git://github.com/asdf-vm/asdf.git",
             has_downloads: true,
             has_issues: true,
             has_pages: true,
             has_projects: true,
             has_wiki: true,
             homepage: "https://asdf-vm.com/",
             hooks_url: "https://api.github.com/repos/asdf-vm/asdf/hooks",
             html_url: "https://github.com/asdf-vm/asdf",
             id: 27291507,
             issue_comment_url:
               "https://api.github.com/repos/asdf-vm/asdf/issues/comments{/number}",
             issue_events_url:
               "https://api.github.com/repos/asdf-vm/asdf/issues/events{/number}",
             issues_url: "https://api.github.com/repos/asdf-vm/asdf/issues{/number}",
             keys_url: "https://api.github.com/repos/asdf-vm/asdf/keys{/key_id}",
             labels_url: "https://api.github.com/repos/asdf-vm/asdf/labels{/name}",
             language: "Shell",
             languages_url: "https://api.github.com/repos/asdf-vm/asdf/languages",
             license: {
               key: "mit",
               name: "MIT License",
               spdx_id: "MIT",
               url: "https://api.github.com/licenses/mit"
             },
             merges_url: "https://api.github.com/repos/asdf-vm/asdf/merges",
             milestones_url:
               "https://api.github.com/repos/asdf-vm/asdf/milestones{/number}",
             mirror_url: null,
             name: "asdf",
             node_id: "MDEwOlJlcG9zaXRvcnkyNzI5MTUwNw==",
             notifications_url:
               "https://api.github.com/repos/asdf-vm/asdf/notifications{?since,all,participating}",
             open_issues: 52,
             open_issues_count: 52,
             owner: {
               login: "asdf-vm",
               id: 17869024,
               node_id: "MDEyOk9yZ2FuaXphdGlvbjE3ODY5MDI0"
             },
             private: false,
             pulls_url: "https://api.github.com/repos/asdf-vm/asdf/pulls{/number}",
             pushed_at: "2019-11-29T16:45:42Z",
             releases_url: "https://api.github.com/repos/asdf-vm/asdf/releases{/id}",
             score: 115.17725,
             size: 862,
             ssh_url: "git@github.com:asdf-vm/asdf.git",
             stargazers_count: 6253,
             stargazers_url: "https://api.github.com/repos/asdf-vm/asdf/stargazers",
             statuses_url:
               "https://api.github.com/repos/asdf-vm/asdf/statuses/{sha}",
             subscribers_url:
               "https://api.github.com/repos/asdf-vm/asdf/subscribers",
             subscription_url:
               "https://api.github.com/repos/asdf-vm/asdf/subscription",
             svn_url: "https://github.com/asdf-vm/asdf",
             tags_url: "https://api.github.com/repos/asdf-vm/asdf/tags",
             teams_url: "https://api.github.com/repos/asdf-vm/asdf/teams",
             trees_url: "https://api.github.com/repos/asdf-vm/asdf/git/trees{/sha}",
             updated_at: "2019-11-30T05:54:54Z",
             url: "https://api.github.com/repos/asdf-vm/asdf",
             watchers: 6253,
             watchers_count: 6253
           },
           {
             archive_url:
               "https://api.github.com/repos/asdf-vm/asdf/{archive_format}{/ref}",
             archived: false,
             assignees_url:
               "https://api.github.com/repos/asdf-vm/asdf/assignees{/user}",
             blobs_url: "https://api.github.com/repos/asdf-vm/asdf/git/blobs{/sha}",
             branches_url:
               "https://api.github.com/repos/asdf-vm/asdf/branches{/branch}",
             clone_url: "https://github.com/asdf-vm/asdf.git",
             collaborators_url:
               "https://api.github.com/repos/asdf-vm/asdf/collaborators{/collaborator}",
             comments_url:
               "https://api.github.com/repos/asdf-vm/asdf/comments{/number}",
             commits_url: "https://api.github.com/repos/asdf-vm/asdf/commits{/sha}",
             compare_url:
               "https://api.github.com/repos/asdf-vm/asdf/compare/{base}...{head}",
             contents_url:
               "https://api.github.com/repos/asdf-vm/asdf/contents/{+path}",
             contributors_url:
               "https://api.github.com/repos/asdf-vm/asdf/contributors",
             created_at: "2014-11-29T04:41:43Z",
             default_branch: "master",
             deployments_url:
               "https://api.github.com/repos/asdf-vm/asdf/deployments",
             description:
               "Extendable version manager with support for Ruby, Node.js, Elixir, Erlang & more",
             disabled: false,
             downloads_url: "https://api.github.com/repos/asdf-vm/asdf/downloads",
             events_url: "https://api.github.com/repos/asdf-vm/asdf/events",
             fork: false,
             forks: 284,
             forks_count: 284,
             forks_url: "https://api.github.com/repos/asdf-vm/asdf/forks",
             full_name: "asdf-vm/asdf",
             git_commits_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/commits{/sha}",
             git_refs_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/refs{/sha}",
             git_tags_url:
               "https://api.github.com/repos/asdf-vm/asdf/git/tags{/sha}",
             git_url: "git://github.com/asdf-vm/asdf.git",
             has_downloads: true,
             has_issues: true,
             has_pages: true,
             has_projects: true,
             has_wiki: true,
             homepage: "https://asdf-vm.com/",
             hooks_url: "https://api.github.com/repos/asdf-vm/asdf/hooks",
             html_url: "https://github.com/asdf-vm/asdf",
             id: 27291507,
             issue_comment_url:
               "https://api.github.com/repos/asdf-vm/asdf/issues/comments{/number}",
             issue_events_url:
               "https://api.github.com/repos/asdf-vm/asdf/issues/events{/number}",
             issues_url: "https://api.github.com/repos/asdf-vm/asdf/issues{/number}",
             keys_url: "https://api.github.com/repos/asdf-vm/asdf/keys{/key_id}",
             labels_url: "https://api.github.com/repos/asdf-vm/asdf/labels{/name}",
             language: "Shell",
             languages_url: "https://api.github.com/repos/asdf-vm/asdf/languages",
             license: {
               key: "mit",
               name: "MIT License",
               spdx_id: "MIT",
               url: "https://api.github.com/licenses/mit"
             },
             merges_url: "https://api.github.com/repos/asdf-vm/asdf/merges",
             milestones_url:
               "https://api.github.com/repos/asdf-vm/asdf/milestones{/number}",
             mirror_url: null,
             name: "asdf",
             node_id: "MDEwOlJlcG9zaXRvcnkyNzI5MTUwNw==",
             notifications_url:
               "https://api.github.com/repos/asdf-vm/asdf/notifications{?since,all,participating}",
             open_issues: 52,
             open_issues_count: 52,
             owner: {
               login: "asdf-vm",
               id: 17869024,
               node_id: "MDEyOk9yZ2FuaXphdGlvbjE3ODY5MDI0"
             },
             private: false,
             pulls_url: "https://api.github.com/repos/asdf-vm/asdf/pulls{/number}",
             pushed_at: "2019-11-29T16:45:42Z",
             releases_url: "https://api.github.com/repos/asdf-vm/asdf/releases{/id}",
             score: 115.17725,
             size: 862,
             ssh_url: "git@github.com:asdf-vm/asdf.git",
             stargazers_count: 6253,
             stargazers_url: "https://api.github.com/repos/asdf-vm/asdf/stargazers",
             statuses_url:
               "https://api.github.com/repos/asdf-vm/asdf/statuses/{sha}",
             subscribers_url:
               "https://api.github.com/repos/asdf-vm/asdf/subscribers",
             subscription_url:
               "https://api.github.com/repos/asdf-vm/asdf/subscription",
             svn_url: "https://github.com/asdf-vm/asdf",
             tags_url: "https://api.github.com/repos/asdf-vm/asdf/tags",
             teams_url: "https://api.github.com/repos/asdf-vm/asdf/teams",
             trees_url: "https://api.github.com/repos/asdf-vm/asdf/git/trees{/sha}",
             updated_at: "2019-11-30T05:54:54Z",
             url: "https://api.github.com/repos/asdf-vm/asdf",
             watchers: 6253,
             watchers_count: 6253
           }
         ],
         mockFetch: () => console.log("fetch")
       }

export const repositoryDetailsApi = () => ({
  mockFetch: () => console.log("fetch")
})
