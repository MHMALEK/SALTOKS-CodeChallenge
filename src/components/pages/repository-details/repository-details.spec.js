import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {repositoryDetailsApi} from '../../../../__mocks__/server-repsonse';
import RepositoryDetailsPageDumb from './repository-details-page.dumb';
describe('SnappShot for Repository details', () => {
  const mockProps = {
    isLoading: true,
    repositoryData: {},
    readMeUrl:
      'https://github.com/MHMALEK/service-worker-examples/blob/master/README.md',
    readmeContent: 'c2FsdG8ga3MgY29kZSBjaGFsbGVuZ2U=',
    requestRepositoryDetails: () => repositoryDetailsApi.mockFetch(),
  };
  test('snapshot renders', () => {
    const component = renderer.create(
        <MemoryRouter>
          <RepositoryDetailsPageDumb {...mockProps} />
        </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
