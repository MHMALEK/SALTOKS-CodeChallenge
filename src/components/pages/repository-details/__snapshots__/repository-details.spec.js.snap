// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnappShot for Repository details snapshot renders 1`] = `
<div
  className="container mx-auto"
>
  <div
    className="max-w-sm w-full lg:max-w-full lg:flex border shadow min-h-screen mt-6 mb-6"
  >
    <div />
    <p>
      dasd
    </p>
  </div>
</div>
`;
